build/ci/pr/Jenkinsfile

```diff
@@ -20,6 +20,20 @@ pipeline {
     }
 
     stages {
+        stage('Check if Docker image needs rebuilding') {
+            when {
+                expression {
+                    ifImageNeedsRebuilding()
+                }
+            }
+            steps {
+                sh """
+                    export HASH=\$(md5sum operators/Gopkg.lock build/ci/Dockerfile local-volume/Gopkg.lock | awk '{print \$1}' | md5sum | awk '{print \$1}')
+                    export IMAGE=push.docker.elastic.co/eck/eck-ci:\${HASH}
+                    make -C build/ci ci-build-image
+                """
+            }
+        }
```

Comment:
It seems to me we could just `docker pull ... || (docker build ... && docker push ...)` in the `ci-build-image` target and remove this ifImageNeedsRebuilding func and the `export` to just have this stage?

```groovy
stage('Maybe rebuild CI Docker image') {
    steps {
        sh """
            make -C build/ci ci-build-image
        """
    }
}
```

Reply:
Fixed